# About 

This repo houses data and code for the analyses in the publication "Epigenetic attenuation of interferon signaling drives aging-related improvements in systemic lupus". The analyses explores the relationship between aging and gene expression profiles in patients part of the California Lupus Epidemiology Study (CLUES) cohort. It analyzes both single cell PBMC data and bulk RNA-seq data. These results are compared with analysis done on healthy controls from the Rotterdam Study (bulk microarray) and from UCSF (PBMC). Full analysis details are available in the publication Methods section. 



# Files

## Datasets

The tarball Data.tar.gz contains the following files:

- `cleaned_CLUES_metadata.csv`: Metadata for the CLUES bulk cohort (271 observations of 5 variables: age, race/ethnicity, sex, patient IDs, and SLEDAI score) after cleaning and merging with post-QC bulk RNA-seq CLUES data.

- `cleaned_CLUES_bulkcounts.csv`: Bulk RNA-seq CLUES data after quality control, with genes identified by their ENSEMBL ID. Counts are unnormalized.

- `bulk_cohort_metadata.csv`: Metadata for the CLUES bulk cohort with steroid information (271 observations of 8 variables: age, race, ethnicity, patient IDs, sex, and steroid dosage at two different timepoints).

- `ImmGeneCounts.csv`: Gene counts for the CLUES bulk cohort with steroid information, genes identified by their ENSEMBL ID. Counts are unnormalized.

- `raw_slopes.csv`: Unnormalized cell proportion vs. age slope data generated by `Proportion_Slope_Calculation_CLUES-2024.ipynb`.

- `pathway_dotplot.xlsx`: GSEA results for select pathways for cell types of interest, generated from dataframes created in `pseudobulk_analysis.R`.

- `new_files/`: This folder contains the unnormalized pseudobulk counts for each cell type, after pseudobulk-level QC. Also contains metadata for 149 CLUES patients and 49 healthy controls.

## Scripts

- `CLUES_bulk.R`: Takes in `cleaned_CLUES_bulkcounts.csv` and `cleaned_CLUES_metadata.csv` as inputs. Calculates SLEDAI correlation with age, normalizes and conducts differential expression analysis using voom, creates a volcano plot, regression plots for genes of interest, and a pathway plot based on GSEA results using fGSEA and the reactome pathways.

- `steroid_analysis_CLUES_bulk.R`: Takes in `bulk_cohort_metadata.csv` and `ImmGeneCounts.csv` as inputs. Performs QC on the gene counts and cleans the metadata to select patients with accurate steroid information. Normalizes and conducts differential expression analysis using voom, creates a volcano plot, regression plots for genes of interest, and a pathway plot based on GSEA results using fGSEA and the reactome pathways.

- `fine_analysis_age-2024.ipynb`: Takes in single-cell PBMC data. Cleans the data, removes unwanted samples, creates UMAPs of interest, calculates cell type proportion for each cell type, performs the pseudobulk calculation for each cell type for each patient and for each patient across cell types, and creates the `new_files` folder with the data described earlier in this README.

- `pseudobulk_analysis.R`: Analyzes the files in `new_files`. Performs QC on the pseudobulk file for each cell type, normalizes and conducts differential expression analysis using voom, computes GSEA results using fGSEA after calling topTable, and creates pathway plots for certain cell types using the reactome pathways. Populates information in the final dataframes for each cell type to `pathway_dotplot.xlsx`.

- `dotplot.R`: Takes in `pathway_dotplot.xlsx` as the input and creates a dotplot based on the pathways and cell types of interest.

- `Proportion_Slope_Calculation_CLUES-2024.ipynb`: Takes in the local file created by `fine_analysis_age-2024.ipynb`. Calculates the slope for this data with respect to a one unit increase in age for each cell type, calculates and adjusts the p-value for each slope, and displays this result in a dataframe.

- `single_cell_proportion_analysis.R`: Takes in the results from `Proportion_Slope_Calculation_CLUES-2024.ipynb` in the format of a csv (`raw_slopes.csv`). Plots this data in two separate bar plots, one for CLUES and one for healthy controls.

- `supp8.R`: Takes in pseudobulk metadata from the `new_files` folder. Additionally reads in pseudobulk counts for each patient. Generates the pathway and volcano figures for supplementary figure 8.

- `Supp4.R`: Takes in `cleaned_CLUES_bulkcounts.csv` and `cleaned_CLUES_metadata.csv` as inputs. Subsets to female patients. Creates pathway plot based on GSEA results using fGSEA and the reactome pathways.

**R Session Info**

```
version 4.3.1 (2023-06-16) 

Platform: aarch64-apple-darwin20 (64-bit) 

Running under: macOS Sonoma 14.0 

  

Matrix products: default 

BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib  

LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0 

  

locale: 

[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8 

  

time zone: America/Los_Angeles 

tzcode source: internal 

  

attached base packages: 

[1] stats4    stats     graphics  grDevices utils     datasets  methods   base      

  

other attached packages: 

[1] readxl_1.4.3                svglite_2.1.2               purrr_1.0.2                 ggpubr_0.6.0                

[5] janitor_2.2.0               org.Hs.eg.db_3.18.0         AnnotationDbi_1.64.0        RColorBrewer_1.1-3          

[9] stringr_1.5.0               data.table_1.14.8           GEOquery_2.70.0             ggeffects_1.3.2             

[13] ggrepel_0.9.4               ggplot2_3.4.4               fgsea_1.28.0                tidyr_1.3.0                 

[17] DESeq2_1.42.0               SummarizedExperiment_1.32.0 Biobase_2.62.0              MatrixGenerics_1.14.0       

[21] matrixStats_1.0.0           GenomicRanges_1.54.0        GenomeInfoDb_1.38.0         IRanges_2.36.0              

[25] S4Vectors_0.40.1            BiocGenerics_0.48.0         limma_3.58.0                tibble_3.2.1                

[29] dplyr_1.1.3                 

  

loaded via a namespace (and not attached): 

  [1] splines_4.3.1            BiocIO_1.12.0            bitops_1.0-7             ggplotify_0.1.2          filelock_1.0.2           

  [6] cellranger_1.1.0         R.oo_1.25.0              polyclip_1.10-6          XML_3.99-0.14            lifecycle_1.0.3  
  
  [11] rstatix_0.7.2            edgeR_4.0.0              lattice_0.21-8           ensembldb_2.26.0         MASS_7.3-60              

[16] insight_0.19.6           backports_1.4.1          magrittr_2.0.3           yaml_2.3.7               cowplot_1.1.1            

[21] DBI_1.1.3                lubridate_1.9.3          abind_1.4-5              zlibbioc_1.48.0          R.utils_2.12.2           

[26] AnnotationFilter_1.26.0  ggraph_2.1.0             msigdbr_7.5.1            RCurl_1.98-1.12          yulab.utils_0.1.4        

[31] tweenr_2.0.2             rappdirs_0.3.3           GenomeInfoDbData_1.2.11  enrichplot_1.22.0        tidytree_0.4.6           

[36] codetools_0.2-19         DelayedArray_0.28.0      DOSE_3.28.2              xml2_1.3.5               ggforce_0.4.1            

[41] tidyselect_1.2.0         aplot_0.2.2              farver_2.1.1             viridis_0.6.4            BiocFileCache_2.10.1     

[46] GenomicAlignments_1.38.2 jsonlite_1.8.7           tidygraph_1.3.1          systemfonts_1.0.5        tools_4.3.1              

[51] progress_1.2.2           treeio_1.26.0            ragg_1.2.6               Rcpp_1.0.11              glue_1.6.2               

[56] gridExtra_2.3            SparseArray_1.2.0        mgcv_1.8-42              qvalue_2.34.0            withr_2.5.2              

[61] fastmap_1.1.1            fansi_1.0.5              digest_0.6.33            timechange_0.2.0         R6_2.5.1                 

[66] gridGraphics_0.5-1       textshaping_0.3.7        colorspace_2.1-0         GO.db_3.18.0             biomaRt_2.58.2           

[71] RSQLite_2.3.2            R.methodsS3_1.8.2        utf8_1.2.4               generics_0.1.3           rtracklayer_1.62.0       

[76] prettyunits_1.2.0        graphlayouts_1.1.0       httr_1.4.7               S4Arrays_1.2.0           scatterpie_0.2.1         

[81] pkgconfig_2.0.3          gtable_0.3.4             blob_1.2.4               XVector_0.42.0           clusterProfiler_4.10.0   

[86] shadowtext_0.1.3         carData_3.0-5            ProtGenerics_1.34.0      scales_1.2.1             png_0.1-8                

[91] snakecase_0.11.1         ggfun_0.1.4              rstudioapi_0.15.0        tzdb_0.4.0               reshape2_1.4.4           

[96] rjson_0.2.21             nlme_3.1-162             curl_5.1.0               cachem_1.0.8             parallel_4.3.1           

[101] HDO.db_0.99.1            restfulr_0.0.15          pillar_1.9.0             grid_4.3.1               vctrs_0.6.4              

[106] car_3.1-2                dbplyr_2.4.0             readr_2.1.4              GenomicFeatures_1.54.3   cli_3.6.1                

[111] locfit_1.5-9.8           compiler_4.3.1           Rsamtools_2.18.0         rlang_1.1.1              crayon_1.5.2             

[116] ggsignif_0.6.4           labeling_0.4.3           plyr_1.8.9               fs_1.6.3                 stringi_1.7.12           

[121] viridisLite_0.4.2        BiocParallel_1.36.0      babelgene_22.9           munsell_0.5.0            Biostrings_2.70.1        

[126] lazyeval_0.2.2           GOSemSim_2.28.1          Matrix_1.6-3             hms_1.1.3                patchwork_1.1.3          

[131] bit64_4.0.5              KEGGREST_1.42.0          statmod_1.5.0            broom_1.0.5              igraph_2.0.1.1           

[136] memoise_2.0.1            ggtree_3.10.0            fastmatch_1.1-4          bit_4.0.5                ape_5.7-1                

[141] gson_0.1.0   

```



